clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1111,6 +1112,87 @@ public void testSubscriptionChangesWithAutoCommitDisabled() {
         consumer.close();
     }
 
+    @Test
+    public void testUnsubscribeWillTriggerPartitionsRevokeWithValidGeneration() {
+        Time time = new MockTime();
+        SubscriptionState subscription = new SubscriptionState(new LogContext(), OffsetResetStrategy.EARLIEST);
+        ConsumerMetadata metadata = createMetadata(subscription);
+        MockClient client = new MockClient(time, metadata);
+
+        initMetadata(client, Collections.singletonMap(topic, 1));
+        Node node = metadata.fetch().nodes().get(0);
+
+        ConsumerPartitionAssignor assignor = new RangeAssignor();
+        KafkaConsumer<String, String> consumer = newConsumer(time, client, subscription, metadata, assignor, false, groupInstanceId);
+
+        initializeSubscription(consumer, getExceptionConsumerRebalanceListener());
+
+        prepareRebalance(client, node, assignor, singletonList(tp0), null);
+
+        try {
+            consumer.updateAssignmentMetadataIfNeeded(time.timer(Long.MAX_VALUE));
+            consumer.poll(Duration.ZERO);
+            fail("Should throw exception");
+        } catch (Throwable e) {
+            assertEquals(partitionAssigned + singleTopicPartition, e.getCause().getMessage());
+        }
```

Comment:
Use `RuntimeException e = assertThrows(RuntimeException.class, () -> consumer.poll(Duration.ZERO));` instead of try-catch-fail

Reply:
Can you clarify the test intention either by comment or through the name? The test name suggests we expect to successfully invoke the `onPartitionsRevoked` callback, but we actually expect an exception to be thrown. Also, we seem to be checking against the `partitionsAssigned` message, and not the `partitionsRevoked`?